module/sell.py

```diff
@@ -19,32 +20,34 @@ def sell(update: Update, context: CallbackContext) -> None:
                     price = str(format(float(message.split()[2].replace(',', '.')), '.2f'))
                     context.bot.send_message(chat_id, "Ricerca del libro associato all'ISBN inserito...")
                     
-                    books = get_books()
-                    found, n = find_book(user_isbn, books)
-                    if found:
-                        isbn = str(books.iloc[n][0])
-                        title  = str(books.iloc[n][1])
-                        authors  = str(books.iloc[n][2])
-                        title_and_authors  = title + "di" + authors
-                        context.bot.send_message(chat_id, "Il libro è: \nTitolo: \"" + title_and_authors + "\"")
-                        add_item(isbn, title, authors, username, price)
-                        context.bot.send_message(chat_id, "Il libro è stato aggiunto al database.")
-                        return
+                    conn = create_connection(DB_PATH)
```

Comment:
every time you do this, you are opening a new connection with the sqlite database, are you sure about this? When you'll close the connection?

I am really afraid about this, for the dmi bot we open and close the connection to avoid leaking opened connection, so multiple threads, I strongly recommend you to delegate this "conn" inside the add_item and add_book functions.

What do you think?  @TendTo @aegroto

Reply:
Yep, the connection [should be closed](https://docs.python.org/3/library/sqlite3.html). I gueses the idea is to keep a connection always open, but I think it's not a very good practice